test/sql/cast/struct_to_map.test

```diff
@@ -0,0 +1,328 @@
+query I
+SELECT {"name": 'Alice', "age": NULL}::MAP(VARCHAR, VARCHAR);
+----
+{name=Alice, age=NULL}
+
+# Multiple NULL values
+query I
+SELECT {"a": NULL, "b": NULL, "c": 'value'}::MAP(VARCHAR, VARCHAR);
+----
+{a=NULL, b=NULL, c=value}
+
+# Constant vector tests
+
+# Single constant struct
+query I
+SELECT x::MAP(VARCHAR, INT) FROM (SELECT {"constant": 42} as x FROM range(5));
+----
+{constant=42}
+{constant=42}
+{constant=42}
+{constant=42}
+{constant=42}
+
+# Complex nested type tests
+
+# STRUCT values to STRUCT values
+query I
+SELECT {"person": {"name": 'Alice', "age": 25}}::MAP(VARCHAR, STRUCT(name VARCHAR, age INT));
+----
+{person={'name': Alice, 'age': 25}}
+
+# LIST values to LIST values
+query I
+SELECT {"numbers": [1,2,3], "letters": ['a','b']}::MAP(VARCHAR, VARCHAR[]);
+----
+{numbers=[1, 2, 3], letters=[a, b]}
+
+# Nested MAP values
+query I
+SELECT {meta: MAP{'key': 'value'}}::MAP(VARCHAR, MAP(VARCHAR, VARCHAR));
+----
+{meta={key=value}}
+
+# Complex nested structure
+query I
+SELECT col::MAP(VARCHAR, MAP(VARCHAR, INT))
+FROM VALUES ({ data: MAP{'count': 10, 'total': 100} })
+AS tab(col);
+----
+{data={count=10, total=100}}
+
+# Edge cases
+
+# Empty struct (if supported)
+# Note: This might fail depending on how empty structs are handled
+# query I
+# SELECT {}::MAP(VARCHAR, INT);
+# ----
+# {}
+
+# Single field struct
+query I
+SELECT {"single": 'value'}::MAP(VARCHAR, VARCHAR);
+----
+{single=value}
+
+# Large number of fields
+query I
+SELECT {
+    "f1": 1, "f2": 2, "f3": 3, "f4": 4, "f5": 5,
+    "f6": 6, "f7": 7, "f8": 8, "f9": 9, "f10": 10
+}::MAP(VARCHAR, INT);
+----
+{f1=1, f2=2, f3=3, f4=4, f5=5, f6=6, f7=7, f8=8, f9=9, f10=10}
+
+# Error cases
+
+# Cannot cast unnamed struct to MAP
+statement error
+SELECT (1, 2, 3)::MAP(INT, INT);
+----
+Cannot cast unnamed STRUCTs to MAP
+
+# Key casting can fail
+statement error
+SELECT col::MAP(INT, INT)
+FROM VALUES ({"duck": 1}),({"goose": 2})
+AS tab(col);
+----
+Conversion Error: Could not convert string 'duck' to INT32
+
+# Value casting can fail  
+statement error
+SELECT {"number": 'not_a_number'}::MAP(VARCHAR, INT);
+----
+Conversion Error: Could not convert string 'not_a_number' to INT32
+
+# Special character handling in keys
+
+# Keys with spaces
+query I
+SELECT {"first name": 'Alice', "last name": 'Smith'}::MAP(VARCHAR, VARCHAR);
+----
+{first name=Alice, last name=Smith}
+
+# Keys with special characters
+query I
+SELECT {"key-with-dashes": 1, "key_with_underscores": 2, "key.with.dots": 3}::MAP(VARCHAR, INT);
+----
+{key-with-dashes=1, key_with_underscores=2, key.with.dots=3}
+
+# Unicode keys
+query I
+SELECT {"测试": 'test', "🦆": 'duck'}::MAP(VARCHAR, VARCHAR);
+----
+{测试=test, 🦆=duck}
+
+# Case sensitivity preservation
+query I
+SELECT {"CamelCase": 1, "lowercase": 2, "UPPERCASE": 3}::MAP(VARCHAR, INT);
+----
+{CamelCase=1, lowercase=2, UPPERCASE=3}
+
+# Performance and batching tests
+
+# Large batch of rows
+query I
+SELECT col::MAP(VARCHAR, INT)
```

Comment:
Can we have a variation where we keep the FROM but the select is: `SELECT {"value": 42}`
I think this should create a constant vector, which I suspect could cause an issue

Reply:
Then I guess we still allocate for STANDARD_VECTOR_SIZE, probably just leaves the existing buffer untouched when we change to a ConstantVector :thinking:

Reply:
I've added the test and checked with `-DSTANDARD_VECTOR_SIZE=2` and unenven row amounts. Seems to work.